### PR TITLE
feat(rds): performance insights for DatabaseCluster instances

### LINKED ISSUE #7957: Add EnablePerformanceInsights to InstanceProps type
Currently, there is no way to specify the EnablePerformanceInsights value for instances within a DatabaseCluster type (defaults to false). The alternative is to use the low-level Cfn types instead.

I am proposing supporting this configuration parameter on the InstanceProps type.

### Use Case

I need to use this configuration parameter as part of a DatabaseCluster type, and I do not want to have to revert to using CfnDBCluster.

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by njlynch**
Adds the EnablePerformanceInsights and related props to `InstanceProps` for
instances within a cluster.

_Note:_ I opted not to try to coalesce `InstanceProps`,
`DatabaseInstanceNewProps`, and `DatabaseInstanceSourceProps` in this PR; there
are a ton of overlapping properties, but it's not immediately clear which fields
are relevant for cluster instances vs standalone instances. I think
investigating and validiting how to combine these is a significantly larger
task.

fixes #7957

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 1]
```diff
@@ -595,6 +596,13 @@ export class DatabaseCluster extends DatabaseClusterBase {
         // Instance properties
         dbInstanceClass: databaseInstanceType(instanceType),
         publiclyAccessible,
+        enablePerformanceInsights: instanceProps.enablePerformanceInsights,
+        performanceInsightsKmsKeyId: instanceProps.enablePerformanceInsights
+          ? instanceProps.performanceInsightEncryptionKey?.keyArn
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts:601]: I wonder - should passing a `performanceInsightEncryptionKey` imply `enablePerformanceInsights` to `true`? Or, alternatively, should we validate that if `performanceInsightEncryptionKey` was passed, then `enablePerformanceInsights` is `true`? Just some food for thought.

File: packages/@aws-cdk/aws-rds/lib/props.ts [hunk 2]
```diff
@@ -42,6 +42,27 @@ export interface InstanceProps {
...
+   * @default false
+   */
+  readonly enablePerformanceInsights?: boolean;
+
+  /**
+   * The amount of time, in days, to retain Performance Insights data.
+   *
+   * @default 7
+   */
+  readonly performanceInsightRetention?: PerformanceInsightRetention;
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/props.ts:58]: I think we should mention here that this is ignored currently unless `enablePerformanceInsights` is `true`.

File: packages/@aws-cdk/aws-rds/lib/props.ts [hunk 3]
```diff
@@ -42,6 +42,27 @@ export interface InstanceProps {
...
+
+  /**
+   * The AWS KMS key for encryption of Performance Insights data.
+   *
+   * @default - default master key
+   */
+  readonly performanceInsightEncryptionKey?: kms.IKey;
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/props.ts:65]: Ditto - we should mention here that this is ignored currently unless `enablePerformanceInsights` is `true`.

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 4]
```diff
@@ -595,6 +596,13 @@ export class DatabaseCluster extends DatabaseClusterBase {
...
+          : undefined,
+        performanceInsightsRetentionPeriod: instanceProps.enablePerformanceInsights
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts:603]: Same comment here as for `performanceInsightEncryptionKey` 🙂.

File: packages/@aws-cdk/aws-rds/lib/cluster.ts (hunk 1 above)
Comment (njlynch) [reply packages/@aws-cdk/aws-rds/lib/cluster.ts:601]: Yeah, I definitely had that thought too. I ended up opting for "keep existing behavior", but very much willing to improve things instead. I think auto-enabling performance insights if either the key or retention period are specified makes sense, and erroring if those fields are provided with performance insights explicitly disabled. I'll update to that effect.

File: packages/@aws-cdk/aws-rds/test/test.instance.ts [hunk 5]
```diff
@@ -911,4 +912,55 @@ export = {
 
     test.done();
   },
+
+  'performance insights': {
+    'cluster with all performance insights properties'(test: Test) {
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/test/test.instance.ts:917]:
```suggestion
    'instance with all performance insights properties'(test: Test) {
```

**Pull request merged**